scripts/west_commands/build_helpers.py

```diff
@@ -25,21 +27,42 @@
 used; otherwise, it's "{}".'''.format(
     DEFAULT_BUILD_DIR)
 
+def _resolve_build_dir(fmt, cwd, **kwargs):
+    # Check if source_dir is below cwd first
+    source_dir = kwargs.get('source_dir')
+    if source_dir:
+        if escapes_directory(cwd, source_dir):
+            kwargs['source_dir'] = os.path.relpath(source_dir, cwd)
+        else:
+            # no meaningful relative path possible
+            kwargs['source_dir'] = ''
+    try:
+        return fmt.format(**kwargs)
+    except KeyError as k:
+        return None
```

Comment:
Slightly more pythonic would be to let the exception bubble up and handle it in find_build_dir